torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -627,12 +625,20 @@ void ScriptModuleSerializer::writeByteCode(
     const bool save_mobile_debug_info) {
   std::vector<c10::IValue> elements;
   BackendDebugInfoRecorder debug_info_recorder;
-  elements.emplace_back(
-      static_cast<int64_t>(caffe2::serialize::kProducedBytecodeVersion));
+  int64_t version_to_write = caffe2::serialize::kProducedBytecodeVersion;
+
+  // When backporting from v6 to v5 through byte-code re-emitting, the version
+  // should be v5 instead of v6. Another option is to keep v6 here, but in the
+  // backport function, change version number in both bytecode.pkl,
+  // and other files containling the version number, like
+  // mobile_debug_handles.pkl.
+  if (version_to_write == 6 && BytecodeEmitDefaultInputsMode::is_enabled()) {
```

Comment:
Ideally it would be good to read the bytecode.pkl (v6) and just rewrite the version number to v5. If it's urgent, let's follow up on it.

Reply:
yeah this should be part of the backport code not in the serialization code.
Basically the output of a backport function should be a model in the previous version, and all the logic to make that happen that's versioning specific (like setting the right version)  should be self contained in the function.

Reply:
+1 on what Raziel said.

Also, it seems dangerous to check `version_to_write` against magic number `6`. In the future, when byte code version bumps to 7, this code will not be triggered correctly.

Reply:
Thanks! I'll make a util function just to alter the version number.

Reply:
I have tried a few options (https://fb.quip.com/MNuMAOgiHQdT), but have not succeeded in just updating the version number in bytecode.pkl. The fundamental reason is that we rely on writeArchive() to write the .pkl file, which is stateful. It depends on the constant metadata which was written before, which is related to the tensor's storageImple pointer. It's different from time to time for each write. 

It's also coupled with another state, type_name_uniquer_ to resolve type name collisions.

Reply:
@iseeyuan Updated the document at https://fb.quip.com/MNuMAOgiHQdT#DVLACAoItek with some py code to update just the bytecode in the pickle. Can be done in C++ too.